gcc-4.8/post-link.sh

```diff
@@ -34,19 +34,50 @@ if [ "$(uname)" != "Darwin" ]; then
         # Locate the system's C-runtime object files and link them into the gcc 
         #  build so they are automatically on the gcc search path.
         # (The location of these files varies from one system to the next.)
-        C_RUNTIME_OBJ_FILES="crt0.o crt1.o crt2.o crt3.o crti.o crtn.o"    
-
-        for obj_file in $C_RUNTIME_OBJ_FILES; do
-            for libdir in `/sbin/ldconfig -v 2> /dev/null | grep '^/' | awk '{print $1}'`; do
-                # Remove trailing ':'
-                libdir=${libdir:0:${#libdir}-1}
-                if [ -e $libdir/$obj_file ]; then
-                    ln -s $libdir/$obj_file ${PREFIX}/lib/gcc/*/*/
-                    break
+        C_RUNTIME_OBJ_FILES="crt0.o crt1.o crt2.o crt3.o crti.o crtn.o"
+
+        c_runtime_obj_files_found=0
+
+        # Try locating crtXXX.o in default library search paths
+        for library_path in $(ld --verbose | grep SEARCH_DIR | sed -r 's/SEARCH_DIR\("=?([^"]*)"\);/ \1/g'); do
+            for obj_file in $C_RUNTIME_OBJ_FILES; do
+                obj_file_full_path="$library_path/$obj_file"
+                if [[ -e "$obj_file_full_path" ]]; then
+                    ln -s "$obj_file_full_path" "${PREFIX}/lib/gcc/"*/*/
+                    c_runtime_obj_files_found=1
                 fi
             done
+            if [ $c_runtime_obj_files_found -eq 1 ]; then
+                break
```

Comment:
It won't stop after the first file, rather after the first folder with one or more crtXXX.o files in it! See the order of `for`s